⃣### Fase 1: "Doctor por un día"

● Se seleccionan 5-10 muestras de pacientes del dataset sin mostrar sitienen diabetes.

● Cada equipo (o alumno) analiza los valores (glucosa, IMC, presión, etc.) y decide si el paciente tiene o no diabetes.

● Se anotan las predicciones en una pizarra o tabla.

In [6]:
import pandas as pd
import numpy as np
from sklearn.impute import SimpleImputer
from sklearn.preprocessing import OneHotEncoder
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score
from sklearn.ensemble import RandomForestClassifier

In [7]:
df = pd.read_csv('datos_sin_diabetes.csv')
df.head(10)


Embarazos  Glucosa  Presion_Arterial  Espesor_Piel  Test Insulina   IMC  \
0          6      148                72            35              0  33.6   
1          1       85                66            29              0  26.6   
2          8      183                64             0              0  23.3   
3          1       89                66            23             94  28.1   
4          0      137                40            35            168  43.1   
5          5      116                74             0              0  25.6   
6          3       78                50            32             88  31.0   
7         10      115                 0             0              0  35.3   
8          2      197                70            45            543  30.5   
9          8      125                96             0              0   0.0   

   DiabetesPedigree  Edad  
0             0.627    50  
1             0.351    31  
2             0.672    32  
3             0.167    21  
4             2.288    33  
5             0.201    30  
6             0.248    26  
7             0.134    29  
8             0.158    53  
9             0.232    54

In [8]:
muestras = df.sample(n=10, random_state=42)
print(muestras)

     Embarazos  Glucosa  Presion_Arterial  Espesor_Piel  Test Insulina   IMC  \
668          6       98                58            33            190  34.0   
324          2      112                75            32              0  35.7   
624          2      108                64             0              0  30.8   
690          8      107                80             0              0  24.6   
473          7      136                90             0              0  29.9   
204          6      103                72            32            190  37.7   
97           1       71                48            18             76  20.4   
336          0      117                 0             0              0  33.8   
568          4      154                72            29            126  31.3   
148          5      147                78             0              0  33.7   

     DiabetesPedigree  Edad  
668             0.430    43  
324             0.148    21  
624             0.158    21  

Los pacientes 336( por la herencia y por el alto nivel de azucar), 473 por glucosa, 568, 148 por lo mismo.

In [9]:
#Preparar los datos 

#Separamos las características y la variable objetivo
x = df.drop(columns =['diabetes'], axis=1)
y = df['diabetes']

#dividimos los datos en entrenamiento y prueba
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=42)

#nomalizamos los datos

sc= StandardScaler() #creamos una instancia del objeto StandarScaler
x_train = sc.fit_transform(x_train) #Calculamos la media y la desviación estándar y normalizamos los datos de entrenamiento
x_test = sc.transform(x_test) #Aplicamos la transformación a los datos de prueba



KeyError: "['diabetes'] not found in axis"

In [ ]:
#Entrenamos el modelo
modelo = RandomForestClassifier(n_estimators=100, random_state=42) #Creamos una instancia del modelo
modelo.fit(x_train, y_train) #Entrenamos el modelo

#Hacemos predicciones
y_pred = modelo.predict(x_test) 

#Evaluamos el modelo

accuracy = accuracy_score(y_test, y_pred)
matriz_confusion = confusion_matrix(y_test, y_pred)

print('Exactitud:', accuracy)
print('Matriz de confusión:\n', matriz_confusion)



NameError: name 'RandomForestClassifier' is not defined